File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int p2 = numbers.size() - 1;
        int sum;
        for(int p1=0;p1<p2;){
            sum = numbers[p1]+numbers[p2];
            if(sum == target){
                return {p1+1,p2+1};
            }else if(sum>target)
                p2--;
            else
                p1++;
        }
        return {};
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        p1 = 0
        p2 = len(numbers) - 1
        while p1<p2:
            sum_ = numbers[p1]+numbers[p2]
            if sum_ > target:
                p2 -= 1
            elif sum_ < target:
                p1 += 1
            else:
                return [p1+1,p2+1]